src/integrationTest/groovy/UmplePluginIntegrationTests.groovy

```diff
@@ -0,0 +1,88 @@
+import org.junit.*
+import static org.junit.Assert.*
+import org.gradle.testfixtures.ProjectBuilder
+import org.gradle.api.Project
+import org.gradle.tooling.* //GradleConnector, ProjectConnection, BuildLauncher
+
+public class UmpleIntegrationTests {
+    private static Project testProject
+    private static final DIRECTORIES_TO_CHECK = 
+    [
+        "src${File.separator}integrationTest${File.separator}resources${File.separator}TestProject${File.separator}subproj${File.separator}build${File.separator}classes${File.separator}generatedSource",
+        "src${File.separator}integrationTest${File.separator}resources${File.separator}TestProject${File.separator}subproj${File.separator}generated${File.separator}java",
+        "src${File.separator}integrationTest${File.separator}resources${File.separator}TestProject${File.separator}sub2${File.separator}build${File.separator}classes${File.separator}generatedSource",
+        "src${File.separator}integrationTest${File.separator}resources${File.separator}TestProject${File.separator}sub2${File.separator}generated${File.separator}java"
+    ] 
+    private static final PATH_TO_TEST_PROJECT = "src${File.separator}integrationTest${File.separator}resources${File.separator}TestProject"
+    private static final TEST_PROJECT_TASK = "compileJava"
+
+    @Before
+    public void setUp() {
+        testProject = ProjectBuilder.builder().build()
+        testProject.plugins.apply 'umple.gradle.plugin'
+    }
+
+    @Test
+    public void applyUmplePluginToProject() {      
+        assertTrue('Umple plugin was not applied', testProject.getPlugins().hasPlugin('umple.gradle.plugin'))
+    }
+
+    @Test
+    public void umplePluginAddsTaskToProject() {
+        assertTrue('generateSource task was not added to project', testProject.tasks.getByName('generateSource') != null)
+    }
+    
+    // Try generating Java from master.ump and compiling the result
+    @Test
+    public void generateAndCompileJavaFiles() {        
+        GradleConnector connector = GradleConnector.newConnector()
+        connector.forProjectDirectory(new File(PATH_TO_TEST_PROJECT))
+        ProjectConnection connection = connector.connect()
+        try {
+            BuildLauncher launcher = connection.newBuild()
+            launcher.forTasks(TEST_PROJECT_TASK)
+            launcher.run()
+        } finally {
+            connection.close()
+        }
+        
+        for (String current : DIRECTORIES_TO_CHECK) {
```

Comment:
This set of checks isn't good enough. We need to check that the files are properly generated. For example, check the main repository and rip out one of the simpler test cases. Then, check the tested code matches the expected. 

Unfortunately, this introduces some dependencies on Umple itself. I wonder if we should be extracting some of the test resources into a "fixture project." This side discussion is not relevant to this PR, though. Just copy a test case or two of content out and make sure the files are generated properly.